games/maze/src/maze.rs

```diff
@@ -292,4 +326,50 @@ pub mod tests {
             assert_eq!(serialized.as_str(), expected);
         }
     }
+
+    #[test]
+    /// test
+    fn neighbouring_tile_types_shows_correct_tile_types() {
+        let test_cases = vec![
+            /// Player on the outer left edge
+            NeighbouringTileTypesTestCase {
+                size: 2,
+                player_position: Position { x: 0, y: 0 },
+                compare_tile_types: NeighbouringTileTypes {
+                    left: TileType::Blocked,
+                    right: TileType::Open,
+                    up: TileType::Blocked,
+                    down: TileType::Blocked,
+                },
+            },
+            /// Player in the middle
+            NeighbouringTileTypesTestCase {
+                size: 3,
+                player_position: Position { x: 1, y: 1 },
+                compare_tile_types: NeighbouringTileTypes {
+                    left: TileType::Blocked,
+                    right: TileType::Open,
+                    up: TileType::Open,
+                    down: TileType::Blocked,
+                },
+            },
+            /// Player on the bottom right edge
+            NeighbouringTileTypesTestCase {
+                size: 100,
+                player_position: Position { x: 99, y: 99 },
+                compare_tile_types: NeighbouringTileTypes {
+                    left: TileType::Blocked,
+                    right: TileType::Blocked,
+                    up: TileType::Open,
+                    down: TileType::Blocked,
+                },
+            },
+        ];
+        for case in test_cases {
+            let mut maze = Maze::new(case.size);
```

Comment:
You probably don't want to use Maze::new() for this, it will be a random maze at somepoint and you don't want random output in the maze. This is what I created the `maze_from_slice_with_player_at(x, y, map_slice)` function for.